### DISCUSSION TITLE
[Вопрос] Переименование ботов + ДК

### DISCUSSION THREAD

**Discussion by xBarselonx** [trickerer/Trinity-Bots#232] [Q&A]
trickerer, добрый день, скажи пожалуйста можно ли переименовать бота который создан командой .npcbot createnew, каким-то способом? Я пробовал их переименовать в ручную в базе в таблице creature_template_npcbot_appearance (если не ошибаюсь в названии таблицы, что это та таблица где есть записи ника бота, пол, цвет волос и т.д) но в игре ничего не поменялось. 
Также при создании рыцаря смерти через команду .npcbot createnew выводится сообщение типа расса не соответствует классу. Раз не получилось создать через команду решил это сделать в ручную в базах creature_template_npcbot_appearance и creature_template_npcbot_extras, но позже в игре используя команды  .npcbot lookup и .npcbot spawn в списке созданных рыцарей смерти не оказалось. 
В качестве программы для работы с базами данных использую HeidiSQL.

**Comment by trickerer** [comment]
1. Проблемы переименования описаны в https://github.com/trickerer/Trinity-Bots/issues/194. Если коротко - переименовывать через таблицу бесполезно (ну или малоэффективно).
2. Про ДК: ![ncre](https://user-images.githubusercontent.com/1254989/198566592-b0eee698-dea6-4ce6-8fab-aa8c976b060d.jpg)

P.S. имена юнитов хранятся в `creature_template`

**Reply by xBarselonx** [reply]
Я так понимаю ботов рыцарей смерти нужно создавать в окне запуска worldserver.exe? Но не в клиенте игры? 

Что касается переименования ботов, в моем случае лучше удалить ботов в ручную в базе и создать новых. Если да, то скаких таблиц лучше начать?

**Comment by trickerer** [comment]
1. Нет разницы. Нужно проверять правильность параметров комманды.
2. Если в клиенте закэшировано имя юнита с ID 70800, то удаление бота из таблицы и создание нового с тем же ID ничего не даст. Нужно очищать кэш клиента. Если нужно почистить нестандартных ботов, то в мануале указаны таблицы, в которых хранятся все данные. Созданные боты имеют ID 70800-70999. https://github.com/trickerer/Trinity-Bots#database

**Reply by xBarselonx** [reply]
> * Нет разницы. Нужно проверять правильность параметров комманды.
> * Если в клиенте закэшировано имя юнита с ID 70800, то удаление бота из таблицы и создание нового с тем же ID ничего не даст. Нужно очищать кэш клиента. Если нужно почистить нестандартных ботов, то в мануале указаны таблицы, в которых хранятся все данные. Созданные боты имеют ID 70800-70999. https://github.com/trickerer/Trinity-Bots#database

Все получилось. Спасибо!
